vfs/memdb/api.go

```diff
@@ -66,3 +69,18 @@ func Delete(name string) {
 	defer memoryMtx.Unlock()
 	delete(memoryDBs, name)
 }
+
+// TestDB creates an empty shared memory database for the test to use.
+// The database is automatically deleted when the test and all its subtests complete.
+// Each subsequent call to TestDB returns a unique database.
+func TestDB(tb testing.TB) string {
+	tb.Helper()
+	name := fmt.Sprintf("%s_%p", tb.Name(), tb)
+	tb.Cleanup(func() { Delete(name) })
+	Create(name, nil)
+	return (&url.URL{
+		Scheme:   "file",
+		OmitHost: true,
+		Path:     "/" + name,
+	}).String() + "?vfs=memdb&"
```

Comment:
You can always tack it onto the returned string afterwards but it's a bit awkward.

Reply:
None of this is an accident.

Using `url.URL` is the only way for me to [encode the path](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/url/url.go;l=131-136) (not as a path segment).

Because some common subtest names have `/`s (which I'd rather not encode) and others have `#`s (which I must encode).

I didn't use `url.Values` because I wanted the final `&`, which makes it easier to do:
```go
memdb.TestDB(t) + "_pragma=busy_timeout(10000)"
```

But your idea is a good one too.

Reply:
Went with your proposal.